lib/my_app.rb

```diff
@@ -0,0 +1,151 @@
+ENV['RACK_ENV'] ||= "development"
+
+require_relative '../data_mapper_setup.rb'
+require 'sinatra/base'
+require 'sinatra/flash'
+
+class MakersBnb < Sinatra::Base
+  use Rack::MethodOverride
+  register Sinatra::Flash
+
+  enable :sessions
+  set :session_secret, 'super secret'
+
+  helpers do
+    def current_user
+      @current_user ||= User.get(session[:user_id]) 
+    end
+  end
+
+  get '/' do
+    @spaces = Space.all
+    erb :index
+  end
+
+  post '/' do
+    a_space = Space.create(params)  
+    current_user.space << a_space
+    current_user.save
+    redirect to('/new')
+  end
+
+  get '/new' do 
+    erb :newPost
+  end
+
+  get '/users/new' do
+    @user = User.new
+    erb :new_user
+  end
+
+  post '/users' do
+    @user = User.create(email: params[:email],
+                        password: params[:password],
+                        password_confirmation: params[:password_confirmation],
+                        user_name: params[:user_name])
```

Comment:
You could just pass `params` here, as the hash you're passing to `create` is exactly the same.